src/modules/cpu/host_tensor_augmentations.hpp

```diff
@@ -9191,4 +9191,1741 @@ RppStatus gridmask_i8_i8_host_tensor(Rpp8s *srcPtr,
+                    {
+                        srcPtrTempR -= vectorIncrementPerChannel;
+                        srcPtrTempG -= vectorIncrementPerChannel;
+                        srcPtrTempB -= vectorIncrementPerChannel;
+
+                        __m256 p[6];
+                        rpp_simd_load(rpp_load48_u8pln3_to_f32pln3_mirror_avx, srcPtrTempR, srcPtrTempG, srcPtrTempB, p);    // simd loads
+                        compute_cmn_24_host(p[0], p[1], p[2], pCMNParams);  // cmn adjustment
+                        compute_cmn_24_host(p[3], p[4], p[5], pCMNParams);  // cmn adjustment
+                        rpp_simd_store(rpp_store48_f32pln3_to_u8pkd3_avx, dstPtrTemp, p);    // simd stores
+   
+                        dstPtrTemp += vectorIncrement;
+                    }
+                    for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                    {
+                        srcPtrTempR--;
+                        srcPtrTempG--;
+                        srcPtrTempB--;
+
+                        dstPtrTemp[0] = (Rpp8u) RPPPIXELCHECK((((Rpp32f) (*srcPtrTempR)) - mean) * stdDev);
+                        dstPtrTemp[1] = (Rpp8u) RPPPIXELCHECK((((Rpp32f) (*srcPtrTempG)) - mean) * stdDev);
+                        dstPtrTemp[2] = (Rpp8u) RPPPIXELCHECK((((Rpp32f) (*srcPtrTempB)) - mean) * stdDev);
+
+                        dstPtrTemp += 3;
+                    }
+
+                    srcPtrRowR += srcDescPtr->strides.hStride;
+                    srcPtrRowG += srcDescPtr->strides.hStride;
+                    srcPtrRowB += srcDescPtr->strides.hStride;
+                    dstPtrRow += dstDescPtr->strides.hStride;
+                }
+            }
+        }
+
+        // Crop Mirror Normalize without fused output-layout toggle (NHWC -> NHWC or NCHW -> NCHW)
+        else
+        {
+            if(mirrorFlag == 0)
+            {
+                Rpp32u alignedLength = (bufferLength / 16) * 16;
+                srcPtrChannel = srcPtrImage + (roiPtr->xywhROI.xy.y * srcDescPtr->strides.hStride) + (roiPtr->xywhROI.xy.x * layoutParams.bufferMultiplier);
+                for(int c = 0; c < layoutParams.channelParam; c++)
+                {
+                    Rpp8u *srcPtrRow, *dstPtrRow;
+                    srcPtrRow = srcPtrChannel;
+                    dstPtrRow = dstPtrChannel;
+
+                    for(int i = 0; i < roiPtr->xywhROI.roiHeight; i++)
+                    {
+                        Rpp8u *srcPtrTemp, *dstPtrTemp;
+                        srcPtrTemp = srcPtrRow;
+                        dstPtrTemp = dstPtrRow;
+
+                        int vectorLoopCount = 0;
+                        for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                        {
+                            __m256 p[2];
+
+                            rpp_simd_load(rpp_load16_u8_to_f32_avx, srcPtrTemp, p);    // simd loads
```

Comment:
You could use memcpy for this branch since there is no channel toggle or mirroring involved. Check performance.

Reply:
We are subtracting the mean and dividing by stdDev here. We cannot directly do memcpy here i guess?